ModelLibrary/Process/BankStatementPayment.cs

```diff
@@ -213,53 +249,223 @@ private String CreatePayment(MBankStatementLine bsl)
+                else if (_ds != null && _ds.Tables[0].Rows.Count > 1)
                 {
-                    payment.SetC_Currency_ID(invoice.GetC_Currency_ID());
-                    payment.SetPayAmt(invoice.GetGrandTotal(true));
+                    if (!payment.Save(Get_Trx()))
+                    {
+                        Get_Trx().Rollback();
+                        return null;
+                    }
+                    else
+                    {
+                        for (int i = 0; i < _ds.Tables[0].Rows.Count; i++)
+                        {
+
+                            MPaymentAllocate PayAlocate = new MPaymentAllocate(GetCtx(), 0, Get_Trx());
+                            PayAlocate.SetC_Payment_ID(payment.GetC_Payment_ID());
+                            PayAlocate.SetAD_Client_ID(payment.GetAD_Client_ID());
+                            //set Organization with the reference of Bank Account
+                            PayAlocate.SetAD_Org_ID(invoice.GetAD_Org_ID());
+                            PayAlocate.SetC_Invoice_ID(invoice.GetC_Invoice_ID());
+                            PayAlocate.SetC_InvoicePaySchedule_ID(Util.GetValueOfInt(_ds.Tables[0].Rows[i]["C_INVOICEPAYSCHEDULE_ID"]));
+
+                            if (invoice.IsSOTrx())
+                            {
+                                PayAlocate.SetInvoiceAmt(Util.GetValueOfDecimal(_ds.Tables[0].Rows[i]["DUEAMT"]));
+                                PayAlocate.SetAmount(Util.GetValueOfDecimal(_ds.Tables[0].Rows[i]["DUEAMT"]));
+                            }
+                            else
+                            {
+                                PayAlocate.SetInvoiceAmt(-1 * Util.GetValueOfDecimal(_ds.Tables[0].Rows[i]["DUEAMT"]));
+                                PayAlocate.SetAmount(-1 * Util.GetValueOfDecimal(_ds.Tables[0].Rows[i]["DUEAMT"]));
+                            }
+                            if (!PayAlocate.Save(Get_Trx()))
+                            {
+                                Get_Trx().Rollback();
+                                ValueNamePair pp = VLogger.RetrieveError();
+                                string error = pp != null ? pp.GetValue() : "";
+                                if (string.IsNullOrEmpty(error))
+                                {
+                                    error = pp != null ? pp.GetName() : "";
+                                    if (string.IsNullOrEmpty(error))
+                                    {
+                                        error = pp != null ? pp.ToString() : "";
+                                    }
+                                }
+                                _message = !string.IsNullOrEmpty(error) ? error : "VA012_PaymentNotSaved";
+                                return null;
+                            }
+                            else
+                            {
+                                payment.SetPayAmt(payment.GetPayAmt() + PayAlocate.GetAmount());
```

Comment:
this else section not required, system will do it on after save functionality

Reply:
removed else section!